src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -685,6 +675,52 @@ private void getCanvasResponse(HttpServletRequest request,
 		out.flush();
 	}
 
+	//When there is too much data to retrieve at once data will be paged.
+	//When data is paged Link Headers will be send in http response. We are
+	//only concerned with the Next Link Header. This method will extract header 
+	//if one exists.
+	private String extractNextLink(HttpResponse canvasResponse) {
+		String linkValueString = null;
+		String linkValueNext = null;
+		String searchPhrase = "rel=\"next\"";
+		M_log.debug("SEARCH PHRASE: " + searchPhrase);
+		HashMap<String, String> links = new HashMap<String, String>();
+		Header[] headers = canvasResponse.getAllHeaders();
```

Comment:
So, rather than getting all the headers and looping through them until you find the one named "Link", can't you simply ask for the "Link" header and do something if it exists?

For example:

* [`HttpResponse.getFirstHeader(String name)`](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpMessage.html#getFirstHeader(java.lang.String)) (or `HttpResponse.getLastHeader(String name)`):  Returns the first (or last) header with a specific name.
* [`HttpResponse.getHeaders(String name)`](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpMessage.html#getHeaders(java.lang.String)): If you need to check every occurrence of the header, returns an array of only the headers with the name you want.
* [`HttpResponse.headerIterator(String name)`](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpMessage.html#headerIterator(java.lang.String)): Like `HttpResponse.getHeaders()`, but returns an iterator.  It may be more compact.
* [`HttpResponse.containsHeader(String name)`](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpMessage.html#containsHeader(java.lang.String)): Might be useful, but since the other methods handle the case of the missing header, it's probably not necessary.

Some headers are allowed to appear multiple times, which is why these deal with multiples.  If you think "Link" may appear more than once, `HttpResponse.headerIterator()` is the safer bet.